docs/source/keys.rst

```diff
@@ -0,0 +1,293 @@
+	// Bob's public key
+	const kBpub = pubkey:03a5aded4cfa04cb4b49d4b19fe8fac0b58802983018cdd895a28b643e7510c1fb
+	//Carl's public key
+	const kCpub = pubkey:03bd94ee8e570da8815f5660bab86aca010d950ddfb87458bb0dcafbc8ea6f9657
+
+	//versig used to model  a *2-of-3* multi signature schema
+	transaction T {
+		input = _
+		output = 1BTC: fun(x, y). versig(kApub, kBpub, kCpub; x, y)
+	}
+
+
+In this case, output script ``versig(kApub, kBpub, kCpub; x, y)`` evaluates to true
+if the two parameters can match two of the three required keys.
+For instance, let ``sigC`` be Carl's signature  and ``sigB`` Bob's signature, then
+``versig(kApub, kBpub, kCpub; sigB, sigC)`` evaluates to ``true`` but
+``versig(kApub, kBpub, kCpub; sigC, sigB)`` does not. 
+
+
+
+""""""""
+Signing
+""""""""
+
+The :ref:`function <label_c_functions>` ``sig`` generates signatures.
+There are two constructs for it:
+
+-  ``sig(k)`` is  used inside a transaction and generates the signature of the transaction itself. In this case, the signature is *lazy* and is generated when compiling the transaction.
+- ``sig(k) of T`` generates the signature of transaction ``T`` using private key ``k``. It cannot be used inside the definition of a  transaction.
+
+
+The signing operation signs all the fields of the transaction *but*
+the witness fields.  This is necessary not to incurr in
+infinite signatures, and it allows multiple signatures to be added
+without invalidating previous ones.
+   
+Consider the following transaction ``A_funds``, redeemable with 
+a signature made by Alice.
+
+.. code-block:: btm
+		
+    // Alice's public key
+    const kApub = pubkey:033806fe039c8d149c8e68f4665b4a479acab773b20bddf7df0df59ba4f0567341
+
+    // tx redeemable with Alice's private key
+    transaction A_funds {input = _ output = 1BTC: fun(x). versig(kApub; x)}
+
+
+Let us define transaction ``TA``   to redeem transaction ``A_funds``, like this:
+   
+.. code-block:: btm
+
+	//Alice's private key	
+	const kA = key:cSFqKAaCUGHZoGDrcreo3saZpMv9NvcVmRZVbVddbodEuzWNCDNt
+
+	// defining  transaction TA
+	transaction TA {
+		input = A_funds : sig(kA)   //signature of the transaction
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+Alternatively, it is possible to use the other construct,  like this:
+
+.. code-block:: btm
+
+	//Alice's private key	
+	const kA = key:cSFqKAaCUGHZoGDrcreo3saZpMv9NvcVmRZVbVddbodEuzWNCDNt
+
+	// transaction with witness not specified
+	transaction T {
+		input = A_funds : _ 
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+	// signing transaction T
+	const sigA = sig(kA) of T 
+
+	// defining the actual transaction TA,
+	//with the same fields as in T so to use  signature sigA
+	transaction TA {
+		input = A_funds : sigA 
+		output = 1BTC: fun(x). versig(kApub; x) //any condition 
+	}
+
+
+Transaction ``TA`` uses as witness the signature obtained by ``T``,
+which has same input and output fields.
```

Comment:
Transaction ``TA`` uses as witness the signature obtained by ``T``, which has same input and output fields.
--> 
The signature of `TA_noWitness` and `TA` is the same, since they differ only by the witness field.
Even if it works, **this approach is strongly discouraged**, since it can be confusing, duplicates the code,
and lead to poor maintainability. A better approach is presented below.